app/_includes/head.html

```diff
@@ -158,6 +158,13 @@
     href="/assets/hub.css?v={{ site.time | date: '%s' }}"
   />
   {% endif %}
+
+  {% if page.no_version != true %}
+  {% if page.kong_version == site.data.kong_latest.release %}<link rel="canonical" href="{{ site.links.web }}{{ page.url }}" />{% else %}<link rel="canonical" href="{{ site.links.web }}{% assign latest_page_url = page.url | replace: page.kong_version, page.kong_latest.release %}{% capture latest_page_exists %}{% page_exists {{latest_page_url}} %}FOUR{% endcapture %}{{ latest_page_url }}" />{%endif%}
```

Comment:
Something looks off with the last portion:
```
{% capture latest_page_exists %}{% page_exists {{latest_page_url}} %}FOUR{% endcapture %}{{ latest_page_url }}
```
This is capturing `latest_page_exists` but then the capture isn't used anywhere. As well, `latest_page_url` isn't defined here.

I tested out a couple of older docs that don't have a parallel latest page, and they're indiscriminately pointing to a 2.3.x doc that doesn't exist. For example:

https://deploy-preview-2639--kongdocs.netlify.app/enterprise/0.36-x/getting-started/add-admin/ 
- If you look at the link in the yellow banner, it take you to `/enterprise`. That's because `/enterprise/2.3.x/getting-started/add-admin/` doesn't exist. When you check the header, this non-existent page is set as the canonical url:

![Screen Shot 2021-02-25 at 9 14 33 AM](https://user-images.githubusercontent.com/54370747/109190410-e8faca80-7749-11eb-8a2f-8efb9c4780b6.png)

Reply:
Apologies that this took me a bit but I've leverged `page.exists` in order to avoid applying canonical links to pages that do not resolve to a latest version. 

Unfortunately this means that pages such as [https://deploy-preview-2639--kongdocs.netlify.app/enterprise/0.36-x/getting-started/add-admin/](https://deploy-preview-2639--kongdocs.netlify.app/enterprise/0.36-x/getting-started/add-admin/) will not have a canonical link defined in the page head, so as of now only pages that are either self-referential canonicals or that have canonicals that exist as the latest version will be defined. This should still provide a net benefit regarding SEO and indexing.

Reply:
Hmm. We'll need to find another way to deal with those, then, because they're a big part of what's polluting our search results. Maybe this is a case for `noindex`, then? I can work on that in the future.

Reply:
I'd imagine there's a better way of handling this than running a loop through pages to compare version numbers and ultimately terminating at the greatest available version but I'd have to dig a bit more into the nuances of Liquid, since this ends up in a soul-crushingly slow build time.
I'd hate to go nuclear with `noindex` - hopefully as a half-measure this will at least help clean up some of the results.

Reply:
Yeah, and the loop is used a couple of times in other situations... that can't be good for the build time. If you ever have the time to take a look, the doc team will be eternally grateful 🙏